supervision/detection/utils.py

```diff
@@ -866,6 +866,35 @@ def merge_data(
     return merged_data
 
 
+def merge_metadata(metadata_list: List[Dict[str, Any]]) -> Dict[str, Any]:
```

Comment:
Contrary to how leniently we treat `Detections.empty`, let's be more aggressive here.

Using similar logic to `__eq__`, let's make sure that for all non-empty metadata, all the keys and values are the same.

If it's different, let's raise an error.

Reply:
My goal is to tighten new data structures, while keeping old ones, with known issues, more lenient.